bitcoin/test/fuzz/miniscript_decode.cpp

```diff
@@ -0,0 +1,121 @@
+// Copyright (c) 2019-2021 The Bitcoin Core developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#include <core_io.h>
+#include <hash.h>
+#include <script/miniscript.h>
+#include <script/script.h>
+#include <span.h>
+#include <test/fuzz/FuzzedDataProvider.h>
+#include <test/fuzz/fuzz.h>
+#include <test/fuzz/util.h>
+#include <util/strencodings.h>
+
+
+#include <optional>
+
+/** A class encapulating conversion routines for CPubKey. Since we can't have access to a
+ * hash->pubkey mapping, and since fuzzing the conversion itself here would be useless
+ * computation we use a static dummy public key and its hash. */
+struct MockedKeyConverter {
```

Comment:
Perhaps it's worth having 2 or 3 of each (keys/hashes/...) so that it can test not only script structure, but also that used keys/etc match? This may not matter now, but I expect we may want to extend this test later (post integration into Core) so that the produced satisfactions actually get verified against the script interpreter.

Reply:
After implementing the fuzz target checking against the script interpreter, i modified this using a few dozens precomputed keys, hashes, and dummy signatures. Pretty close to what you did in the unit test setup.

For what it's worth, the commit implementing the satisfaction check against the interpreter on top of this and the signing support is at https://github.com/darosior/bitcoin/commit/ef7c3f0d51c6c92eff400925be7abe8963e05d1e .